packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -794,9 +791,9 @@ export function useGlobalStylesOutput() {
 		return !! getSettings().disableLayoutStyles;
 	} );
 
-	useEffect( () => {
+	return useMemo( () => {
 		if ( ! mergedConfig?.styles || ! mergedConfig?.settings ) {
-			return;
+			return [];
```

Comment:
Does this return statement mean that the whole function could early return an empty array?

It looks like prior to this PR, the function will always return an array with four elements. From looking at the `useState` calls, an equivalent to the default values in that case might be something like:

`return [ [], {}, {}, true ]` ?

Reply:
> Does this return statement mean that the whole function could early return an empty array?

Correct. Initially, I was planning to return the array with the same signature, but then I checked the usage, and returning `undefined` for values won't be an issue.

Reply:
> but then I checked the usage, and returning undefined for values won't be an issue.

Very interesting! Particularly because of `hasBlockGapSupport` having a default truthy value. Given that the function doesn't appear to be exported outside of the `edit-site` package, this sounds okay to me in principle.

One thing to keep in mind with making changes to the output, is that from the documentation it sounds like there might have been (or be) an intention of potentially extracting the global styles APIs to their own package ([this line](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/README.md#L3)). I think I remember @jorgefilipecosta working on making global styles data available in the post editor a while back, but I'm not sure if that's related.

I see that including `hasBlockGapSupport` was added to the output of the function in https://github.com/WordPress/gutenberg/pull/40185 in April. From the looks of it, that value was only needed in `toStyles` and doesn't appear to be used by any consumers of `useGlobalStylesOutput`. It might be worth double-checking with @oandregal to see if there was a reason it was added to the output or whether it's still needed? It looks like we might be able to remove it altogether from the return value 🤔

Reply:
I think this is the best time to change return types if we want 😄 

I was wondering about the `hasBlockGapSupport` return as well. So let'ss see if we can remove it.

Reply:
> I see that including hasBlockGapSupport was added to the output of the function in #40185 in April. From the looks of it, that value was only needed in toStyles and doesn't appear to be used by any consumers of useGlobalStylesOutput.

Yeah, I don't see it in use anywhere either and don't remember why it was returned in the first place :)

Reply:
Thanks for the confirmation, @oandregal. I removed `hasBlockGapSupport` from the return value.